rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/action/TbSaveToCustomCassandraTableNode.java

```diff
@@ -81,16 +88,26 @@ public class TbSaveToCustomCassandraTableNode implements TbNode {
     private PreparedStatement saveStmt;
     private ExecutorService readResultsProcessingExecutor;
     private Map<String, String> fieldsMap;
+    private long ttl;
 
     @Override
     public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
         config = TbNodeUtils.convert(configuration, TbSaveToCustomCassandraTableNodeConfiguration.class);
         cassandraCluster = ctx.getCassandraCluster();
         if (cassandraCluster == null) {
-            throw new RuntimeException("Unable to connect to Cassandra database");
-        } else {
-            startExecutor();
-            saveStmt = getSaveStmt();
+            throw new TbNodeException("Unable to connect to Cassandra database", true);
+        }
+        ctx.addTenantProfileListener(this::onTenantProfileUpdate);
+        onTenantProfileUpdate(ctx.getTenantProfile());
+        startExecutor();
+        saveStmt = getSaveStmt();
+    }
+
+    private void onTenantProfileUpdate(TenantProfile tenantProfile) {
+        DefaultTenantProfileConfiguration configuration = (DefaultTenantProfileConfiguration) tenantProfile.getProfileData().getConfiguration();
+        ttl = config.getDefaultTTL();
+        if (ttl == 0L) {
+            ttl = TimeUnit.DAYS.toSeconds(configuration.getDefaultStorageTtlDays());
```

Comment:
we need to make this backward compatible. If TTL is not set(null) we shouldn't check for value from tenant profile.